Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -801,69 +826,104 @@ public final class ManifestLoader: ManifestLoaderProtocol {
             cmd += ["-o", compiledManifestFile.pathString]
 
             // Compile the manifest.
-            let compilerResult = try Process.popen(arguments: cmd, environment: toolchain.swiftCompilerEnvironment)
-            let compilerOutput = try (compilerResult.utf8Output() + compilerResult.utf8stderrOutput()).spm_chuzzle()
-            result.compilerOutput = compilerOutput
-
-            // Return now if there was an error.
-            if compilerResult.exitStatus != .terminated(code: 0) {
-                return
-            }
-
-            // Pass an open file descriptor of a file to which the JSON representation of the manifest will be written.
-            let jsonOutputFile = tmpDir.appending(component: "\(packageIdentity)-output.json")
-            guard let jsonOutputFileDesc = fopen(jsonOutputFile.pathString, "w") else {
-                throw StringError("couldn't create the manifest's JSON output file")
-            }
+            Process.popen(arguments: cmd, environment: toolchain.swiftCompilerEnvironment) { result in
+                let compilerResult : ProcessResult
+                switch result {
+                case .success(let result):
+                    compilerResult = result
+                case .failure(let error):
+                    return completion(.failure(error))
+                }
```

Comment:
similar to https://github.com/apple/swift-package-manager/pull/3872/files#r750486345

Reply:
Fixed in fd4f7e27.